### ISSUE TITLE
Bootstrapping (using consul)

### ISSUE THREAD

**Issue by zonzamas** [aweber/rabbitmq-autocluster#29]
Im doing tests with a 2 nodes rabbit cluster. Everything works fine except bootstrapping. If I bootstrap (using some custom scripts and puppet to apply the config) both nodes at the same time they never connect to each other. All I can get is a 

    =INFO REPORT==== 23-Oct-2015::10:44:33 ===
    autocluster: Registering node with consul

    =INFO REPORT==== 23-Oct-2015::10:44:33 ===
    autocluster: Node appears to be the first in the cluster

In both nodes. For this no to happen I have to wait for the first node to be up before bootstrapping the second one (restarting any node also works)

Is this an expected behaviour?

**Comment by gmr** [comment]
Hmm looks like you might have run into a race condition on startup. I'm looking to address such a situation in 0.5

**Comment by scalp42** [comment]
Running into the same issue with `dns` backend:

```
=INFO REPORT==== 16-Dec-2015::22:16:03 ===
autocluster: Registering node with dns

=INFO REPORT==== 16-Dec-2015::22:16:03 ===
autocluster: Node appears to be the first in the cluster
```

I set up two nodes, ready to go with rabbitMQ stopped, then started one by one with like 1 min time difference so not sure what's happening.

**Comment by Supernomad** [comment]
This particular issue just bit me as well. Seems like there is no way to get this to work unless you manually start each node individually, which well just plain doesn't work with distributed automation systems.

Any thoughts on how to fix this issue?

**Comment by gmr** [comment]
Logging at the debug level might help in a situation like this.

One scenario I have seen is where the nodes can not communicate with each other, such as a non-matching .erlang.cookie, firewalled ports, or when running in a docker container, not using ``--net=host``.

If it is due to the startup race condition of both starting at the same time, that should be addressed in 0.5.0 which will be released within the next week.
